# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/wr/webrender/res/ps_copy.glsl
# Commit: 8d13c0ab66f0
# Full Hash: 8d13c0ab66f0ac137b1461d3451ecb9361ec2d5e
# Author: Jamie Nicol <jnicol@mozilla.com>
# Date: 2023-03-22 21:15:54
# Regressor Bug: 1823411
# File Overlap Count: 1
# Description:
#   Bug 1823411 - Give all webrender's shader varyings an explicit precision. r=gfx-reviewers,lsalzman
#   
#   Mali profiling tools have shown we are frequently fragment bound, in
#   particular due to varying interpolation. To help mitigate this, we
#   should use mediump where possible. Currently most of our varyings are
# ==============================================================================

diff -r 3919fbd972a6 -r 8d13c0ab66f0 gfx/wr/webrender/res/ps_copy.glsl
--- a/gfx/wr/webrender/res/ps_copy.glsl	Tue Mar 21 19:04:22 2023 +0000
+++ b/gfx/wr/webrender/res/ps_copy.glsl	Tue Mar 21 19:05:19 2023 +0000
@@ -13,7 +13,7 @@
 attribute vec4 a_dst_rect;
 attribute vec2 a_dst_texture_size;
 
-varying vec2 v_uv;
+varying highp vec2 v_uv;
 
 void main(void) {
     // We use texel fetch so v_uv is in unnormalized device space.
@@ -30,7 +30,7 @@
 
 out vec4 oFragColor;
 
-varying vec2 v_uv;
+varying highp vec2 v_uv;
 
 uniform sampler2D sColor0;
 